main.tf

```diff
@@ -166,15 +166,20 @@ module "container_definition" {
       hostPort      = 0
       protocol      = "tcp"
     },
-  ]
+    ], var.enable_prometheus_metrics ? [{
+      containerPort = 9100
+      hostPort      = 0
+      protocol      = "tcp"
+    }] : []
+  )
 
   docker_labels = {
     "traefik.enable"                       = true
     "traefik.http.routers.traefik.rule"    = "Host(`${module.this.name}.${var.domain}`)"
     "traefik.http.routers.traefik.service" = "api@internal"
   }
 
-  command = [
+  command = compact(concat([
```

Comment:
i think we don't need compact here, do we? There no null values in the list possible if i'm not mistaken, as you always either have no additional elements or two.

Reply:
Yeah true we don't, I'll remove compact from there.